app.R

```diff
@@ -150,180 +150,159 @@ shinyApp (
-          #tcga = rbind(tcga,read.table(paste0("https://storage.googleapis.com/tcga_shiny/TCGA_Proteomics_app/Gene_files/TMT/",g,".tsv"),sep="\t",stringsAsFactors=F,header=T,check.names=FALSE))
-          #SELECT Gene,UPN,Value FROM Genes WHERE (Gene='DNMT3A' OR Gene='TP53') AND Type='TMT';
+          query <- paste(query,sep="","Gene='",g,"'"," OR ")
         }
-        
-        # Data cleanup and preparing for output
-        df = melt(tcga,id.vars="gene")
-        names(df) <- c("Gene", "UPN", "Log2.Expression")
-        df <- merge(df, clinicaldata_tmt, by="UPN") # SELECT * FROM TMT_Clinical WHERE (UPN="ND8" OR ...);
+        query <- paste(sep="",substring(query,1, nchar(query)-4),")")
+        query <- paste(query,"AND Type='TMT';") # Make variable for TMT
+        tcga <- dbGetQuery(database,query)
+
+
+        # SELECT * FROM TMT_Clinical WHERE (UPN="ND8" OR ...);
+        query <- "SELECT * FROM TMT_Clinical WHERE ("
+        for(g in unique(tcga$UPN)){
+          query <- paste(query,sep="","UPN='",g,"'"," OR ")
+        }
+        query <- paste(sep="",substring(query,1, nchar(query)-4),");")
+        df <- dbGetQuery(database,query)
+        df <- merge(tcga, df, by="UPN")
         dmtm <<- df
-        
-        g <- ggplot(df,aes(fill=Gene, y=Log2.Expression, x=Name, text = paste0("UPN ID: ",UPN,"<br />TCGA Sample ID: ", TCGA_ID))) + geom_bar(position="dodge", stat="identity") + theme_bw() +
-          theme(text=element_text(size=12, family="avenir", face="bold"), axis.text=element_text(size=10, family="avenir", face="bold"), 
+
+
+        g <- ggplot(df,aes(fill=Gene, y=Value, x=Name, text = paste0("UPN ID: ",UPN,"<br />TCGA Sample ID: ", TCGA_ID))) + geom_bar(position="dodge", stat="identity") + theme_bw() +
+          theme(text=element_text(size=12, family="avenir", face="bold"), axis.text=element_text(size=10, family="avenir", face="bold"),
                 axis.title=element_text(size=12, family="avenir", face="bold"),
                 axis.text.x = element_text(angle = 90, hjust = 1)) +
           ggtitle("Multiple protein view") +
           ylab("Log2 Expression") + xlab("")
         ggplotly(g, tooltip="text")
-        
       }
     })
-    
+
     # Output file
     output$downloadData_t_m <- downloadHandler(
       filename = function() {paste0("Output.csv")},
       content = function(file) {write.csv(dmtm, file, quote = F, row.names = F, sep = ",")}
     )
-    
-    
-    
-    
-    
-    
-    
-    
+
+
+
     # SUBTYPE #
     output$singlegene_plot_subtype_tmt <- renderPlotly({
       # Getting actual genes to plot
-      #genesToPlot <- toupper(input$single_gene_name_2_t)
-      #subtypesToPlot <- input$subtypes_t
-      genesToPlot <- "TP53"
-      subtypesToPlot <- c("M0","M1","M2","M3","M4","M5","Healthy Lin-")
-      tcga = NULL
+      genesToPlot <- toupper(input$single_gene_name_2_t)
```

Comment:
Is the `toupper()` call here so that someone can type "dnmt3a" and will still match "DNMT3A" in the gene list?  Not necessary to fix here, but we should think ahead about how this will interact with mouse genes, which have a mixed-case format (Dnmt3a).  Sai may have already handled this in other apps, I don't know.